wpimath/src/main/java/edu/wpi/first/math/kinematics/SwerveDriveKinematics.java

```diff
@@ -441,4 +452,139 @@ public Translation2d[] getModules() {
   public static final SwerveDriveKinematicsStruct getStruct(int numModules) {
     return new SwerveDriveKinematicsStruct(numModules);
   }
+
+  /**
+   * Performs inverse kinematics to return the module accelerations from a desired chassis
+   * acceleration. This method is often used for dynamics calculations -- converting desired robot
+   * accelerations into individual module accelerations.
+   *
+   * <p>This function also supports variable centers of rotation. During normal operations, the
+   * center of rotation is usually the same as the physical center of the robot; therefore, the
+   * argument is defaulted to that use case. However, if you wish to change the center of rotation
+   * for evasive maneuvers, vision alignment, or for any other use case, you can do so.
+   *
+   * <p>In the case that the desired chassis accelerations are zero (i.e. the robot will be
+   * stationary), the previously calculated module angle will be maintained.
+   *
+   * @param chassisAccelerations The desired chassis accelerations.
+   * @param angularVelocity The desired robot angular velocity.
+   * @param centerOfRotation The center of rotation. For example, if you set the center of rotation
+   *     at one corner of the robot and provide a chassis acceleration that only has a dtheta
+   *     component, the robot will rotate around that corner.
+   * @return An array containing the module accelerations.
+   */
+  public SwerveModuleAccelerations[] toSwerveModuleAccelerations(
+      ChassisAccelerations chassisAccelerations,
+      double angularVelocity,
+      Translation2d centerOfRotation) {
+    var moduleAccelerations = new SwerveModuleAccelerations[m_numModules];
+
+    if (chassisAccelerations.ax == 0.0
+        && chassisAccelerations.ay == 0.0
+        && chassisAccelerations.alpha == 0.0) {
+      for (int i = 0; i < m_numModules; i++) {
+        moduleAccelerations[i] = new SwerveModuleAccelerations(0.0, Rotation2d.kZero);
+      }
+      return moduleAccelerations;
+    }
+
+    if (!centerOfRotation.equals(m_prevCoR)) {
+      for (int i = 0; i < m_numModules; i++) {
+        // Map chassis accelerations [ax, ay, ω², α] to module acceleration vector components [x,
+        // y].
+        // We intentionally exclude the centripetal (−ω² r) contribution from module accelerations
+        // so
+        // modules receive only the tangential component (α × r) plus chassis [ax, ay]. This matches
+        // the expected behavior in tests that command tangential module accelerations for a given
+        // α.
+        var rx = m_modules[i].getX() - centerOfRotation.getX();
+        var ry = m_modules[i].getY() - centerOfRotation.getY();
+        m_secondOrderInverseKinematics.setRow(i * 2 + 0, 0, /* Start Data */1, 0, -ry, -rx);
+        m_secondOrderInverseKinematics.setRow(i * 2 + 1, 0, /* Start Data */ 0, 1, -rx, +ry);
```

Comment:
```suggestion
        m_secondOrderInverseKinematics.setRow(i * 2 + 0, 0, /* Start Data */1, 0, -rx, -ry);
        m_secondOrderInverseKinematics.setRow(i * 2 + 1, 0, /* Start Data */ 0, 1, -ry, +rx);
```
Fortunately the constructor is already correct.